packages/next/server/render.tsx

```diff
@@ -274,6 +277,49 @@ function checkRedirectValues(
   }
 }
 
+// Create the wrapper component for a Flight stream.
+function createServerComponentRenderer(
```

Comment:
We should move this to compile time, but that doesn't block this PR.

Reply:
An overall refactoring to the middleware, `server/next-server` and `server/render` is needed.